### PR TITLE
pchain_network v0.6

### PR DESCRIPTION
**Pull request by ong-jonas**
### Network v0.6 
1. Configuration - Message ID is no longer overridden, back to [rust-libp2p default](https://docs.rs/libp2p/latest/libp2p/gossipsub/struct.Config.html#method.message_id) (“concatenate the source peer id with a sequence number”).
2. Change State Machine Replication Topic hash: 
      "consensus" -> "hotstuff_rs"
      base64url(public_addr) -> "hotstuff_rs/" + base64url(public_addr)
3. Each replica now subscribes to individual topics of closest replicas in kademlia k bucket. However, messages under these topics are not processed and just propagated to the rest of the network via gossipsub. 
4. Dropped Transaction message type removed. 
5. Protocol name is now a configurable string. 
6. Added unsubscribe function to unsubscribe from individual topic of replicas when they disconnect.

### CODE REVIEW

File: src/conversions.rs [hunk 1]
```diff
@@ -133,6 +126,19 @@ pub fn multi_addr(ip_address: Ipv4Addr, port: u16) -> Multiaddr {
     format!("/ip4/{}/tcp/{}", ip_address, port).parse().unwrap()
 }
 
+/// Check the distance between 2 peers. Subscribe to new peer's individual topic
+/// if the distance is below [MAX_REPLICA_DISTANCE]
+pub(crate) fn is_close_peer(peer_1: &PeerId, peer_2: &PeerId) -> bool {
```
Comment (lyulka) [inline src/conversions.rs]: I don’t think this should be in the conversions module.

Any reason why this function should be in the "conversions" module?

File: src/peer.rs
Comment (lyulka) [inline src/peer.rs:1]: I'm looking at `PeerStartError::SystemConfigError`. Its itemdoc says that the variant corresponds to the case where we "Failed to read from system configuration path". However, I don't see anywhere in the code where we read a file at a path.

File: src/conversions.rs (hunk 1 above)
Comment (ong-jonas) [reply src/conversions.rs]: Will shift this to peer module.

File: src/peer.rs
Comment (ong-jonas) [reply src/peer.rs:1]: I've updated it with the correct error variant.

**Review by lyulka — APPROVED** [review]

**Pull request merged**